### PR TITLE
[NFC][CLANG] Fix static analyzer bugs about large copy by values

### PR DESCRIPTION
**Pull request by smanna12**
Reported by Static Analyzer tool:

In getSourceRangeToTokenEnd(clang::Decl const *, clang::SourceManager const &, clang::LangOptions): A very large function call parameter exceeding the high threshold is passed by value

   pass_by_value: Passing parameter LangOpts of type clang::LangOptions (size 1784 bytes) by value, which exceeds the high threshold of 512 bytes

### CODE REVIEW

File: clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ void ModuleDepCollector::addOutputPaths(CowCompilerInvocation &CI,
 }
 
 static CowCompilerInvocation
-makeCommonInvocationForModuleBuild(CompilerInvocation CI) {
+makeCommonInvocationForModuleBuild(const CompilerInvocation &CI) {
```
Comment (Xazax-hun) [inline clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp:113]: This does not compile, `CI` is mutated in this function.

Comment (smanna12) [reply clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp]: >>This does not compile, CI is mutated in this function.
Yes, i missed that. Thank you @Xazax-hun for reviews. 

I have updated patch.

**Review by Xazax-hun — APPROVED** [review]

**Comment by Xazax-hun** [comment]
Could you also update the description of the patch?

**Comment by smanna12** [comment]
> Could you also update the description of the patch?

Done. Thank you @Xazax-hun for reviews!

**Comment by haoNoQ** [comment]
Thanks for catching these!

I wonder what would it take to delete the copy constructors on these classes, so that these mistakes are never made again.

**Review by haoNoQ — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you for reviews everyone!

**Pull request merged**